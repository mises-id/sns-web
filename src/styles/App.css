body{
  background-color: #F8F8F8;
}
iframe{
  background: red;
  left: 9999999999px !important;
}
.launchApp{
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 50px;
  background-color:#EEEFFF;
  z-index: 300;
  padding:10px 15px;
  box-sizing: border-box;
}
.launchApp-logo{
  width: 30px;
  height: 30px;
}
.launchApp-empty{
  height: 50px;
}
.link{
  color: #5D61FF;
  text-decoration: none;
}
.launchApp-txt{
  padding-left: 8px;
  width: 200px;
  font-size: 12px;
  font-family: SF UI Display;
  color: #555555;
  line-height: 16px;
  word-break: break-word;
}
.launchApp-txt span{
  font-weight: bold;
}
.download-btn,.popup-btn{
  width: 83px;
  height: 28px;
  background: #5D61FF;
  border-radius: 14px;
  text-align: center;
  line-height: 28px;
  color: white;
  font-size: 12px;
}
.popup-img{
  display: block;
  margin: 20px auto;
  width: 40px;
}
.popup-txt{
  text-align: center;
  font-weight: bold;
  margin: 0;
}
.popup-btn{
  margin: 20px auto;
}