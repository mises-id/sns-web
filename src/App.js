import React, { useEffect } from 'react';
import './styles/App.css';
import { ConfigProvider} from 'zarm';
import { BrowserRouter as Router, Redirect } from 'react-router-dom'
import enUS from 'zarm/lib/config-provider/locale/en_US';
import 'zarm/dist/zarm.css';
import routeConfig from './router';
import { routes } from './utils/reactUtil';
import { Provider, useDispatch, useSelector } from 'react-redux';
import { persistor, store } from './stores';
import {PersistGate} from 'redux-persist/lib/integration/react';
import { hot } from 'react-hot-loader/root'
import {setTheme} from '@/styles/setZarmTheme'
import { CacheSwitch } from 'react-router-cache-route'
// import { useState } from 'react';
import { useLogin } from './components/PostsIcons/common';
import { setVisibility } from './actions/app';
import { Popup } from 'antd-mobile';
import { isMisesBrowser } from './utils';
const SetRoute = ()=>{
  const {isLogin} = useLogin()
  return <Router>
  <CacheSwitch>
    {routes(routeConfig)}
    <Redirect to={isLogin ? "/home/discover" : "/home/me"} />
  </CacheSwitch>
</Router>
}
const  Download = ()=>{
  // const selector = useSelector((state) => state.user) || {};
  const isMises = isMisesBrowser()
  return !isMises&&!['/download'].includes(window.location.pathname) ? <>
    <div className='launchApp m-flex'>
      <div className='m-flex-1 m-flex'>
        <img src="/static/images/logo.png" alt="" className='launchApp-logo'/>
        <p className='launchApp-txt'>
          <span>Mises </span>
          - A social network protocol based on blockchain technology
        </p>
      </div>
      <div className='download-btn' onClick={()=>{
        window.location.href = '/download'
      }}>Download</div>
    </div>
    <div className='launchApp-empty'></div>
  </> : ''
}
// 封装antd的popup
const DownloadPopUp = ()=>{
  const visible = useSelector(state=>state.app.visible)
  const dispatch = useDispatch()
  return <Popup
    visible={visible}
    onMaskClick={() => {
      dispatch(setVisibility(false))
    }}
    destroyOnClose
    bodyStyle={{
      borderTopLeftRadius: '15px',
      borderTopRightRadius: '15px',
    }}
  >
    <img src="/static/images/logo.png" alt="" className='popup-img'/>
    <p className='popup-txt'>Download Mises to start your Web3 journey</p>
    <div className='popup-btn' onClick={()=>{
        dispatch(setVisibility(false))
        window.location.href = '/download'
      }}>Download</div>
  </Popup>
}
  

const App = ()=> {
  // const [isHref, setisHref] = useState(false)
 
  useEffect(() => {
    setTheme()
    // const error = e=>{
    //   if(e.message==="Uncaught SyntaxError: Unexpected token '<'" 
    //   || (e.reason && ["CSS_CHUNK_LOAD_FAILED","ChunkLoadError"].includes(e.reason.code || e.reason.name))){
    //     const pathname = window.location.pathname
    //     if(!isHref&&pathname!=='error'){
    //       window.location.replace('/error')
    //       setisHref(true)
    //     }
    //   }
    // }
    // window.addEventListener('error',error,true)
    // window.addEventListener('unhandledrejection',error)
    // return ()=>{
    //   window.removeEventListener('error',error)
    //   window.removeEventListener('unhandledrejection',error)
    // }
    // eslint-disable-next-line
  }, [])
  return <ConfigProvider locale={enUS} theme="light" primaryColor='#5c65f6'>
    <Provider store={store}>
      <PersistGate persistor={persistor}>
        <Download />
        <SetRoute />
        <DownloadPopUp />
      </PersistGate>
    </Provider>
  </ConfigProvider>;
}

export default process.env.NODE_ENV === "development" ? hot(App) : App;
