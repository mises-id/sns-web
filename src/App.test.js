import React from 'react';
import App from './App';
import { shallow,configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
configure({ adapter: new Adapter() });
describe('App', () => {
  it('should render correctly', () => {
    const component = shallow(
      <App></App>,
    )
    expect(component).toMatchSnapshot();
  });
});