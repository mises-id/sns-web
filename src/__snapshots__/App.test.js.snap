// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<ConfigProvider
  locale={
    Object {
      "ActionSheet": Object {
        "cancelText": "Cancel",
      },
      "Alert": Object {
        "cancelText": "Close",
      },
      "Confirm": Object {
        "cancelText": "Cancel",
        "okText": "OK",
      },
      "DatePicker": Object {
        "cancelText": "Cancel",
        "okText": "OK",
        "title": "please select",
      },
      "DatePickerView": Object {
        "day": "",
        "hour": "",
        "minute": "",
        "month": "",
        "year": "",
      },
      "DateSelect": Object {
        "placeholder": "please select",
      },
      "ImagePreview": Object {
        "loadBefore": "load origin",
        "loadEnd": "loading completed",
        "loadStart": "loading",
      },
      "Keyboard": Object {
        "okText": "OK",
      },
      "Picker": Object {
        "cancelText": "Cancel",
        "okText": "OK",
        "title": "please select",
      },
      "Pull": Object {
        "completeText": "loading completed",
        "dropText": "release to refresh",
        "failureText": "failed to load",
        "loadingText": "loading",
        "pullText": "pull down to refresh",
        "successText": "loaded successfully",
      },
      "SearchBar": Object {
        "cancelText": "Cancel",
        "placeholder": "Search",
      },
      "Select": Object {
        "placeholder": "please select",
      },
      "StackPicker": Object {
        "cancelText": "Cancel",
        "okText": "OK",
      },
      "locale": "en-US",
    }
  }
  primaryColor="#5c65f6"
  theme="light"
>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <PersistGate
      loading={null}
      persistor={
        Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "flush": [Function],
          "getState": [Function],
          "pause": [Function],
          "persist": [Function],
          "purge": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Component>
        <Component>
          <Component
            key="/home"
            path="/home"
            render={[Function]}
          />
          <Component
            key="/userInfo"
            path="/userInfo"
            render={[Function]}
          />
          <Component
            key="/follow"
            path="/follow"
            render={[Function]}
          />
          <Component
            key="/post"
            path="/post"
            render={[Function]}
          />
          <Component
            key="/comment"
            path="/comment"
            render={[Function]}
          />
          <Component
            to="/home"
          />
        </Component>
      </Component>
    </PersistGate>
  </Provider>
</ConfigProvider>
`;
